tests/static_map/heterogeneous_lookup_test.cu

```diff
@@ -64,61 +66,74 @@ struct key_triplet {
 // User-defined device hasher
 struct custom_hasher {
   template <typename CustomKey>
-  __device__ uint32_t operator()(CustomKey const& k)
+  __device__ uint32_t operator()(CustomKey const& k) const
   {
-    return thrust::raw_reference_cast(k).a;
+    return k.a;
   };
 };
 
 // User-defined device key equality
 struct custom_key_equal {
-  template <typename LHS, typename RHS>
-  __device__ bool operator()(LHS const& lhs, RHS const& rhs)
+  template <typename SlotKey, typename InputKey>
+  __device__ bool operator()(SlotKey const& lhs, InputKey const& rhs) const
   {
-    return thrust::raw_reference_cast(lhs).a == thrust::raw_reference_cast(rhs).a;
+    return lhs == rhs.a;
   }
 };
 
-TEMPLATE_TEST_CASE("Heterogeneous lookup",
-                   "",
+TEMPLATE_TEST_CASE_SIG("Heterogeneous lookup",
+                       "",
+                       ((typename T, int CGSize), T, CGSize),
 #if defined(CUCO_HAS_INDEPENDENT_THREADS)  // Key type larger than 8B only supported for sm_70 and
                                            // up
-                   int64_t,
+                       (int64_t, 1),
+                       (int64_t, 2),
 #endif
-                   int32_t)
+
+                       (int32_t, 1),
+                       (int32_t, 2))
 {
-  using Key      = key_pair<TestType>;
-  using Value    = TestType;
-  using ProbeKey = key_triplet<TestType>;
+  using Key        = T;
+  using Value      = T;
+  using InsertKey  = key_pair<T>;
+  using ProbeKey   = key_triplet<T>;
+  using probe_type = cuco::experimental::double_hashing<CGSize, custom_hasher, custom_hasher>;
 
   auto const sentinel_key   = Key{-1};
   auto const sentinel_value = Value{-1};
 
   constexpr std::size_t num      = 100;
   constexpr std::size_t capacity = num * 2;
-  cuco::static_map<Key, Value> map{
-    capacity, cuco::empty_key<Key>{sentinel_key}, cuco::empty_value<Value>{sentinel_value}};
-
-  auto insert_pairs =
-    thrust::make_transform_iterator(thrust::counting_iterator<int>(0),
-                                    [] __device__(auto i) { return cuco::pair<Key, Value>(i, i); });
+  auto const probe               = probe_type{custom_hasher{}, custom_hasher{}};
+  auto my_map                    = cuco::experimental::static_map<Key,
+                                               Value,
+                                               cuco::experimental::extent<std::size_t>,
+                                               cuda::thread_scope_device,
+                                               custom_key_equal,
+                                               probe_type>{capacity,
+                                                           cuco::empty_key<Key>{sentinel_key},
+                                                           cuco::empty_value{sentinel_value},
+                                                           custom_key_equal{},
+                                                           probe};
```

Comment:
```suggestion
  auto my_map                    = cuco::experimental::static_map{capacity,
                                                           cuco::empty_key<Key>{sentinel_key},
                                                           cuco::empty_value<Value>{sentinel_value},
                                                           custom_key_equal{},
                                                           probe};
```
nit: CTAD would work just fine.

Reply:
Beau-ti-ful! Addressed in [cd03620](https://github.com/NVIDIA/cuCollections/pull/381/commits/cd03620c28be7a71bd798081ac9d650166177ae2)